### DISCUSSION TITLE
Need #help with installing rust - Invalid bottle tag symbol

### DISCUSSION THREAD

**Discussion by robni7** [Homebrew/discussions#2429] [Getting started]
Hello, I hope this is a good spot. As the current version of rust does not compile on my 10.11 machine, I want to install al previous version. I read that version 1.10.0 would work fine on my machine.

So I tried installing the formula by running the following command:

`curl https://raw.githubusercontent.com/Homebrew/homebrew-core/595b2e4c8d5fb654c676e61740be97fe1af2795c/Formula/rust.rb > $(find $(brew --repository) -name rust.rb) && brew reinstall rust`

And similar other methods I found on the web. However, I will then get the error

`Error: rust: Invalid bottle tag symbol`

I am not acquainted with how brew works (with bottles etc.), and a strict google search of this message also left me in the dark. I tried 1.9.0 and 1.11.0 commits too to no avail.

So my question is, how do I make it work? If you need more info, please ask. Thanks in advance ;)

**Comment by SMillerDev** [comment]
The syntax of the formula you're trying to install is too old to work with the installed version of brew. You can remove the `bottle do` block and see if you get any further.

**Reply by LeahHarlow** [reply]
Hey! New bootcamp student over here, can I ask what you mean by the bottle do block? I'm having the same issue trying to install postresql.

**Reply by gromgit** [reply]
Open the formula in question with your editor. It will look something like this:
```
class Go < Formula
  desc "Open source programming language to build simple/reliable/efficient software"
  homepage "https://golang.org"
[...]
  bottle do
    sha256 arm64_monterey: "fe137ea6536535299a666ea9e126e84fd58712d25c7071073bb4053da70ad2f0"
    sha256 arm64_big_sur:  "b0a1686c5c5ba668e78a97d66567e9d007f7d2a0c9b1a53e79841e3736d729e6"
    sha256 monterey:       "d40731741ca7d7b16b791fa31d21baab674720c87c00adbf54ca796ff80b9b0f"
    sha256 big_sur:        "01f50ff6dbea579b3b3e792751ec14ba90de1ac5fe09db03adf438558444ddfa"
    sha256 catalina:       "cc6b0bff088b30b80afad3be3519af192e515d83c8928262114ca07e9bf34bd8"
    sha256 mojave:         "a98bd1c5f1cae4907712126d9ea040680084e5fb9743f1d4160744fe3fe8861a"
    sha256 x86_64_linux:   "aa5fbcd2c12be4e08d5edbcbb5eb8a8859c48ecd036c697d2d83b62d410dbd8c"
  end

  # Don't update this unless this version cannot bootstrap the new version.
  resource "gobootstrap" do
[...]
```
That's the `bottle do` block in the middle. Delete the entire thing (including the matching `end` that ends the block).

**Reply by RuKaSuu** [reply]
ok
thanks

**Comment by pestophagous** [comment]
Just wanted to say thanks, because this discussion is still helpful in the year 2023. The discussion here allowed me to get an older formula working on MacOS Ventura 13.3.1 with Homebrew 4.0.13.  The formula that failed was source-highlight/3.1.9_5, but it was able to install and run well after I removed the `bottle do` block from:

/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/source-highlight.rb

Thanks to both the asker and to the respondents here!
